spec/controllers/staff_users_controller_spec.rb

```diff
@@ -28,29 +28,66 @@ def authenticate_staff_user(user)
   end
 
   context "New Account Authentication" do
```

Comment:
This is a controller test, but what's interesting is that we're actually testing domain here, rather than HTTP requests/responses. This might be a sign that we should introduce a service object.